BUILD.md

```diff
@@ -1,66 +1,138 @@
+    below to get a list of supported boards.
 
-Waf should always be called from the ardupilot's root directory.
+    ```sh
+    ./waf configure --board minlure
+    ./waf copter
+    ```
 
-Differently from the make-based build, with Waf there's a configure step
-to choose the board to be used (default is `sitl`):
+    The first command should be called only once or when you want to change a
+    configuration option. One configuration often used is the `--board` option to
+    switch from one board to another one. For example we could switch Pixhawk and
+    build again:
 
-```bash
-# Configure the Linux board
-waf configure --board=linux
-```
+    ```sh
+    ./waf configure --board px4-v2
+    ./waf copter
+    ```
 
-Waf build system is composed of commands. For example, the above command
-(`configure`) is for configuring the build. Consequently, in order to build, a
-"build" command is issued, thus `waf build`. That is the default command, so
-calling just `waf` is enough:
+* **List available boards**
 
-```bash
-# Build programs from bin group
-waf
 
-# Waf also accepts '-j' option to parallelize the build.
-waf -j8
-```
+    It's possible to get a list of supported boards on ArduPilot with the command
+    below
 
-To clean things up, use the `clean` or `distclean` command:
+    ```sh
+    ./waf list_boards
 
-```bash
-# Clean the build products, but keep configure information
-waf clean
+    ```
+
+* **Clean the build**
+
+    Commands `clean` and `distclean` can be used to clean the objects produced by
+    the build. The first keeps the configure information while the second cleans
+    everything.
+
+    Cleaning the build is very often not necessary and discouraged. We do
+    incremental builds reducing the build time by orders of magnitude.
+
+
+* **Upload or install**
+
+    Pixhawk board has and upload option so it uploads to the firmware built to
+    a connected board. In order to use it, pass the option `--upload` to build
+    command:
+
+    ```sh
+    ./waf --targets bin/arducopter-quad --upload
+    ```
+
+    Currently Linux boards don't support the upload option, but there's a install
+    command, which will install to a certain directory. This can be used by
+    distributors to create .deb, .rpm or other package types:
+
+    ```sh
+    ./waf copter
+    DESTDIR=/my/temporary/location ./waf install
+    ```
+
+* **Use different targets**
+
+    The build commands in the items above use `copter` as the target. This
+    builds all binaries that fall under the "copter" group. See the
+    section [Advanced usage](#advanced-usage) below for more details regarding
+    groups.
+
+    This shows a list of all possible targets:
+
+    ```
+    ./waf list
+    ```
+
+    For example, to build only a single binary:
 
-# Clean everything, will need to call configure again
-waf distclean
-```
 
-Using git to clean the files also work fine.
+    ```
+    # Quad frame of ArduCopter
+    ./waf --targets tests/test_own_ptr
```

Comment:
So now test_own_ptr is a build for a quad frame of ArduCopter? What are you trying to do Lucas? Do you want people to crash their drones? :smile: